api/src/main/java/org/openmrs/api/impl/ConditionServiceImpl.java

```diff
@@ -48,6 +48,25 @@ public ConditionDAO getConditionDAO() {
 	 */
 	@Override
 	public Condition saveCondition(Condition condition) {
+		Condition existingCondition = getConditionByUuid(condition.getUuid());
+		if (condition.equals(existingCondition)) {
+			return existingCondition;
+		}
+		if (existingCondition == null) {
+			return conditionDAO.saveCondition(condition);
+		}
+
```

Comment:
Yes I did. I was trying to avoid having to change the original Condition class since it did not have a newInstance method. I could however add it if this is not a problem.

Reply:
You can add that method